OneCommunityGlobal/HighestGoodNetworkApp
# Ilya - Consumables: Single Update

<img width="755" alt="Screenshot 2024-02-29 at 6 17 42 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/105448137/ce62ea76-ad37-486a-8eeb-6df7fea2518b">

# Description
Adds functionality for the user to navigate to [/bmdashboard/consumables](http://localhost:3000/bmdashboard/consumables), select a consumable that needs to have it’s stock updated, press the pencil icon, enter how much of the consumable was used or wasted (in % or units) and press “Update Consumable”. The change should update the Consumables table right away. 

## Related PRS:
To test this frontend PR you need to checkout the [#780](https://github.com/OneCommunityGlobal/HGNRest/pull/780) backend PR.

## Main changes explained:
- New component UpdateConsumable created
- Real-time validation added to warn the user when the quantity entered exceeds current stock
- API calls handled through Redux actions. An action to fetch all consumables is dispatched after the call to update the table with the changes without having to reload the page

## How to test:
1. checkout **ilya_consumables_single_update** branch on the back end. Make sure you are using Node version 14.17.6. Run **npm install** and **npm run build && npm start** to run the backend locally
2. check into current branch on the frontend. Run commands **npm install** and **npm run start:local** to launch the front end application
3. log in as any user
4. go to [http://localhost:3000/bmdashboard](http://localhost:3000/bmdashboard) and log in again
5. go to [http://localhost:3000/bmdashboard/consumables](http://localhost:3000/bmdashboard/consumables), select any item on the table, press the “pencil” icon, enter some numbers next to “Quantity Used” and/or “Quantity Wasted”. Select "%" as units or leave the item's default units 
6. Press “Update Consumable”
7. A success toast message should appear if everything is acceptable or a warning message if not
8. From the Consumables table, press “View” in the “Updates” column and verify that the change you made is reflected. The latest update should be on the bottom

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/83417ee2-5ce8-47ae-84af-fbdf51776d13


## Note:
This change will update the “buildingInventoryItems” collection on the database. As an alternative to step 8 you can open Compass, search the collection by the criteria {"__t":"consumable_item”} and verify that updateRecord property has a new entry with today’s date.
